psydac/fem/basic.py

```diff
@@ -230,19 +230,19 @@ def __getitem__(self, key):
         return self._fields[key]
 
     # ...
-    def __call__( self, *eta ):
+    def __call__( self, *eta , weights=None):
         """Evaluate field at location identified by logical coordinates eta."""
-        return self._space.eval_field( self, *eta )
+        return self._space.eval_field( self, *eta , weights=weights)
 
     # ...
-    def gradient( self, *eta ):
+    def gradient( self, *eta , weights=None):
         """Evaluate gradient of field at location identified by logical coordinates eta."""
-        return self._space.eval_field_gradient( self, *eta )
+        return self._space.eval_field_gradient( self, *eta , weights=weights)
         
     # ...
-    def divergence(self, *eta):
+    def divergence(self, *eta, weights=None):
         """Evaluate divergence of vector field at location identified by logical coordinates eta."""
-        return self._space.eval_field_divergence(self, *eta)
+        return self._space.eval_field_divergence(self, *eta, weights=weights)
```

Comment:
Same comment as above

Reply:
this case is not available yet, so I added a `NotImplementedError`

Reply:
This is the base class, it is not its job to know whether the method exists or not. NotImplementedError is already raised by the subclasses, I believe.

Reply:
done